### PR TITLE
Updated documentation to Doxygen v1.9.2 and new layout.

### PR DESCRIPTION
**Pull request by vovamarch**
Updated Doxygen documentation build to use v1.9.2
Updated layout.
Minor content improvements

### CODE REVIEW

**Review by RobertRostohar — CHANGES_REQUESTED** [review]
In general migration to newer doxygen looks ok.

The `gen_doc.sh` has an issue when called initially (after checkout).
It shows the following error:
`Output directory '../../Documentation/General' does not exist and cannot be created`

`header.html.bak` should be removed.

Now also the pack generator script needs to be updated to use the new `gen_doc.sh`. After that the `genDoc.bat` and `doxygen_general.bat` can be removed,

File: DoxyGen/General/src/introduction.txt [hunk 1]
```diff
@@ -1,43 +1,34 @@
...
+\b ARM.CMSIS-Driver.pdsc  | Package description file in CMSIS-Pack format.
+\b LICENSE.txt            | CMSIS license agreement (Apache 2.0)
+\b Config/                | Configuration files for \ref driver_I2C, \ref driver_NAND, and \ref driver_SPI bus implementations
+\b Documentation/         | This documentation
+\b ETH/                   | \ref driver_eth driver implementations
+\b Flash/                 | \ref driver_Flash memory driver implementations
+\b I2C/                   | \ref driver_I2C driver implementations
+\b NAND/                  | \ref driver_NAND driver implementations
+\b SPI/                   | \ref driver_SPI driver implementations
+\b Utilities/             | Utilities for generating the pack
```
Comment (RobertRostohar) [inline DoxyGen/General/src/introduction.txt]: `Utilities` should be removed

Comment (vovamarch) [reply DoxyGen/General/src/introduction.txt]: Removed with the latest commit.

**Comment by vovamarch** [comment]
Thanks @RobertRostohar!
- updated gen_doc.sh to create the Documentation/ folder if it is  absent.
- removed header.html.bak

I suggest that the pack generation with bash is done separately, and having the documentation working helps with that. Doc publishing can be then also automated with GitHub actions.

**Pull request merged**